lib/community/pages/create_post_page.dart

```diff
@@ -136,15 +137,25 @@ class _CreatePostPageState extends State<CreatePostPage> {
     super.dispose();
   }
 
-  Future<void> _getDataFromLink() async {
-    if (widget.url?.isNotEmpty == true) {
-      final WebInfo info = await LinkPreview.scrapeFromURL(widget.url!);
-      _titleTextController.text = info.title;
+  Future<String?> _getDataFromLink({String? link, bool updateTitleField = true}) async {
+    link ??= widget.url;
+    if (link?.isNotEmpty == true) {
+      try {
+        final WebInfo info = await LinkPreview.scrapeFromURL(link!);
+        if (updateTitleField) {
```

Comment:
Is this parameter being used anywhere? Is it necessary to have it here if it's not being used?

Reply:
Yes. This method was originally called to set the title field. I repurposed it to also be able to return the title, so I wanted to make setting the field optional.

Here's where it's originally called. `updateTitleField` is `true` by default, so no need to pass it.

https://github.com/thunder-app/thunder/blob/774a22a54635d31c21c8669c9102d06398d96f1f/lib/community/pages/create_post_page.dart#L105

And here's where I call it to get the value. I pass `updateTitleField: false` because I don't want it to change anything.

https://github.com/thunder-app/thunder/blob/774a22a54635d31c21c8669c9102d06398d96f1f/lib/community/pages/create_post_page.dart#L273

Hope that makes sense!

Reply:
I see, I missed that. Thanks for the clarification!